crates/autopilot/src/database/auction.rs

```diff
@@ -0,0 +1,161 @@
+use super::Postgres;
+use anyhow::{anyhow, Context, Result};
+use database::{
+    auction::AuctionId,
+    orders::{
+        BuyTokenDestination as DbBuyTokenDestination, OrderKind as DbOrderKind,
+        SellTokenSource as DbSellTokenSource, SigningScheme as DbSigningScheme,
+    },
+    solver_competition::SolverCompetitionId,
+};
+use futures::{StreamExt, TryStreamExt};
+use model::{
+    app_id::AppId,
+    auction::Auction,
+    order::{
+        BuyTokenDestination, Order, OrderData, OrderKind, OrderMetadata, OrderStatus, OrderUid,
+        SellTokenSource,
+    },
+    signature::{Signature, SigningScheme},
+};
+use number_conversions::{big_decimal_to_big_uint, big_decimal_to_u256};
+use primitive_types::H160;
+
+pub struct SolvableOrders {
+    pub orders: Vec<Order>,
+    pub latest_settlement_block: u64,
+}
+
+impl Postgres {
+    pub async fn solvable_orders(&self, min_valid_to: u32) -> Result<SolvableOrders> {
+        let _timer = super::Metrics::get()
+            .database_queries
+            .with_label_values(&["solvable_orders"])
+            .start_timer();
+
+        let mut ex = self.0.begin().await?;
+        let orders = database::orders::solvable_orders(&mut ex, min_valid_to as i64)
+            .map(|result| match result {
+                Ok(order) => full_order_into_model_order(order),
+                Err(err) => Err(anyhow::Error::from(err)),
+            })
+            .try_collect::<Vec<_>>()
+            .await?;
+        let latest_settlement_block =
+            database::orders::latest_settlement_block(&mut ex).await? as u64;
+        Ok(SolvableOrders {
+            orders,
+            latest_settlement_block,
+        })
+    }
+
+    pub async fn replace_current_auction(&self, auction: &Auction) -> Result<AuctionId> {
+        let _timer = super::Metrics::get()
+            .database_queries
+            .with_label_values(&["save_auction"])
+            .start_timer();
+
+        let data = serde_json::to_value(&auction)?;
+        let mut ex = self.0.begin().await?;
+        database::auction::delete_all_auctions(&mut ex).await?;
+        let id = database::auction::save(&mut ex, &data).await?;
+        ex.commit().await?;
+        Ok(id)
+    }
+
+    pub async fn next_solver_competition(&self) -> Result<SolverCompetitionId> {
+        let _timer = super::Metrics::get()
+            .database_queries
+            .with_label_values(&["next_solver_competition"])
+            .start_timer();
+
+        let mut ex = self.0.acquire().await.map_err(anyhow::Error::from)?;
+        database::solver_competition::next_solver_competition(&mut ex)
+            .await
+            .context("failed to get next solver competition ID")
+    }
+}
+
+fn full_order_into_model_order(order: database::orders::FullOrder) -> Result<Order> {
```

Comment:
Since we're here, I would think it doesn't make sense for us to return a full order model with the auction, but instead return something slightly different.

Specifically, I don't need to load all of the metadata.

Reply:
**edit**: Nevermind, you said the same thing:

> I'm planning on eventually changing the Order type that is used in the autopilot and in the auction endpoint to contain only information relevant to solvers so that we don't need this anymore.